src/us/kbase/workspace/database/provenance/CreditMetadata.java

```diff
@@ -0,0 +1,465 @@
+	 *
+	 * @return identifier
+	 */
+	public String getIdentifier() {
+		return identifier;
+	}
+
+	/*
+	 * Gets the license for the resource; may be a string like 'Apache 2.0' or an URL in string
+	 * form.
+	 *
+	 * @return resource license (if present)
+	 */
+	public Optional<String> getLicense() {
+		return Optional.ofNullable(license);
+	}
+
+	/*
+	 * Gets the version information for the resource.
+	 *
+	 * @return resource version (if present)
+	 */
+	public Optional<String> getVersion() {
+		return Optional.ofNullable(version);
+	}
+
+	/*
+	 * Gets the resource type.
+	 *
+	 * @return resource type as a {@link ResourceType}
+	 */
+	public ResourceType getResourceType() {
+		return resourceType;
+	}
+
+	/*
+	 * Gets the comments for the resource.
+	 *
+	 * @return list of comments (strings)
+	 */
+	public List<String> getComments() {
+		return Common.getList(comments);
+	}
+
+	/*
+	 * Gets the contributor list.
+	 *
+	 * @return list of {@link Contributor} objects.
+	 */
+	public List<Contributor> getContributors() {
+		return Common.getList(contributors);
+	}
+
+	/*
+	 * Gets any lifecycle dates.
+	 *
+	 * @return list of {@link EventDate} objects
+	 */
+	public List<EventDate> getDates() {
+		return Common.getList(dates);
+	}
+
+	/*
+	 * Gets the funding information.
+	 *
+	 * @return list of {@link FundingReference} objects
+	 */
+	public List<FundingReference> getFunding() {
+		return Common.getList(funding);
+	}
+
+	/*
+	 * Gets the list of related identifiers.
+	 *
+	 * @return list of {@link PermanentID} objects
+	 */
+	public List<PermanentID> getRelatedIdentifiers() {
+		return Common.getList(relatedIdentifiers);
+	}
+
+	/*
+	 * Gets the list of titles used for the resource.
+	 *
+	 * @return list of {@link Title} objects
+	 */
+	public List<Title> getTitles() {
+		return Common.getList(titles);
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(comments,
+				contributors,
+				dates,
+				funding,
+				identifier,
+				license,
+				relatedIdentifiers,
+				resourceType,
+				titles,
+				version);
+	}
+
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj)
+			return true;
+		if (obj == null)
+			return false;
+		if (getClass() != obj.getClass())
+			return false;
+		CreditMetadata other = (CreditMetadata) obj;
+		return Objects.equals(comments, other.comments)
+				&& Objects.equals(contributors, other.contributors)
+				&& Objects.equals(dates, other.dates)
+				&& Objects.equals(funding, other.funding)
+				&& Objects.equals(identifier, other.identifier)
+				&& Objects.equals(license, other.license)
+				&& Objects.equals(relatedIdentifiers, other.relatedIdentifiers)
+				&& Objects.equals(resourceType, other.resourceType)
+				&& Objects.equals(titles, other.titles)
+				&& Objects.equals(version, other.version);
+	}
+
+	/**
+	 * Gets a builder for the {@link CreditMetadata}.
+	 *
+	 * @param identifier
+	 *                unique persistent ID for the resource
```

Comment:
I've realized I have no idea what the `resource` is in this context or what the ID should look like, if anything. Can you add some docs around that?

Edit: looks like it needs to pass the `checkPID` func.

Reply:
:+1: